src/views/InteractionDesigner.vue

```diff
@@ -207,95 +207,98 @@ export default {
 
     // get the interaction-designer-content positions, will be used to set other elements' position in the canvas (eg: for block editor)
     if (this.activeFlow && this.$refs['interaction-designer-contents'] != undefined) {
-      this.setInteractionDesignerBoundingClientRect(this.$refs['interaction-designer-contents'].getBoundingClientRect())
+      this.setInteractionDesignerBoundingClientRect((this.$refs['interaction-designer-contents'] as Element).getBoundingClientRect())
     }
-  },
-  beforeRouteUpdate(to, from, next) {
+  }
+
+  beforeRouteUpdate(to: Route, from: Route, next: Function) {
     this.activateBlock({blockId: to.params.blockId || null})
     if (to.meta?.isBlockEditorShown) {
       scrollBlockIntoView(to.params.blockId)
       this.setIsBlockEditorOpen(true)
     }
     next()
-  },
-  methods: {
-    ...mapMutations(['deselectBlocks', 'configure']),
-    ...mapMutations('builder', ['activateBlock', 'setIsBlockEditorOpen', 'setInteractionDesignerBoundingClientRect']),
-    ...mapActions('builder', ['setIsEditable']),
-    ...mapMutations('flow', ['flow_setActiveFlowId']),
-
-    ...mapActions([
-      'attemptSaveTree',
-      'discoverTallestBlockForDesignerWorkspaceHeight',
-      'initializeTreeModel']),
-
-    async registerBlockTypes() {
-      const {blockClasses} = this
-
-      const blockInstallers = values(blockClasses).map(async (blockClass) => {
-        const type = blockClass.type
-        const normalizedType = type.replace('.', '_')
-        const typeWithoutSeparators = type.replace('.', '')
-        let {uiComponent} = blockClass
-        if (blockClass.install === undefined) {
-          const exported = await import(`../components/interaction-designer/block-types/${normalizedType}Block.vue`)
-          blockClass.install = exported.install
-          uiComponent = exported.default
-        }
-        invoke(blockClass, 'install', this)
-        Vue.component(`Flow${typeWithoutSeparators}`, uiComponent)
-      })
-      return Promise.all(blockInstallers)
-    },
+  }
 
-    handleCanvasSelected({target}) {
-      if (!target.classList.contains('builder-canvas')) {
-        console.debug('InteractionDesigner', 'Non-canvas selection mitigated')
-        return
+  @Mutation configure: any
```

Comment:
- [ ] use `void` for mutation return type, and check actions' return type from vuex store (most of them could be `void` but not all)